volume/mounts/validate.go

```diff
@@ -26,3 +30,101 @@ func errExtraField(name string) error {
 func errMissingField(name string) error {
 	return errors.Errorf("field %s must not be empty", name)
 }
+
+func lockedPathValidation(path string) error {
+	call, err := middleware.Call("pool.dataset.path_in_locked_datasets", path)
+	if err == nil {
+		if call.(bool) {
+			return errors.Errorf("Dataset path is locked")
+		}
+	}
+	return nil
+}
+
+func isIXVolumePath(path string, datasetPath string) bool {
+	releasePath := filepath.Join("mnt", datasetPath, "releases")
+	if strings.Index(path, "/"+releasePath) == 0 {
+		appPath := strings.Replace(path, "/"+releasePath+"/", "", 1)
+		appName := strings.Split(appPath, "/")[0]
+		volumePath := filepath.Join(releasePath, appName, "volumes", "ix_volumes")
+		if strings.Contains(path, "/"+volumePath) {
+			return true
+		}
+	}
+	return false
+}
+
+func ignorePath(path string) bool {
+	// "/" and "/home/keys/" are added for openebs use only, regular containers can't mount "/" as we have validation
+	// already in place by docker elsewhere to prevent that from happening
```

Comment:
I think there's a docker ticket to change this behavior. May want to check for it and make sure that it has been rejected, and if it hasn't yet, reference the docker ticket here so that we can easily check again in future.